### PR TITLE
feat(core): add Dockerfile and instructions

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,10 +112,45 @@ Then, run the installation and remove the `.deb` file:
...
 dpkg -i /tmp/scw.deb && rm -f /tmp/scw.deb
 ```
+-->
 
 ## With a Docker Image
 
-For each release, we deliver a tagged image on the [Scaleway Docker Hub](https://hub.docker.com/r/scaleway/cli/tags) so can run `scw` in a sandboxed way: _Coming soon..._
+### Manual installation
+
+You can build the `scw` cli with Docker. If you have Docker installed, you can run:
```
Comment (loicbourgois) [inline README.md]:
```suggestion
You can build the `scw` CLI with Docker. If you have Docker installed, you can run:
```

File: README.md [hunk 2]
```diff
@@ -112,10 +112,45 @@ Then, run the installation and remove the `.deb` file:
...
+```sh
+docker build -t scaleway/cli .
+```
+
+You can then use the cli as you would run any image:
```
Comment (loicbourgois) [inline README.md]:
```suggestion
You can then use the CLI as you would run any image:
```

File: README.md [hunk 3]
```diff
@@ -112,10 +112,45 @@ Then, run the installation and remove the `.deb` file:
...
+```
+
+Or if you use ZSH, add the following in your `~/.zshrc`:
+```zsh
+scw() {
+    docker run -i -v $HOME/.config/scw/config.yaml:/.config/scw/config.yaml:ro scaleway/cli $@
+}
+```
+
+You can still use autocompletion while running the cli in Docker, you just need to add the needed command after the definition of the new `scw` function.
```
Comment (loicbourgois) [inline README.md]:
```suggestion
You can still use autocompletion while running the CLI in Docker, you just need to add the needed command after the definition of the new `scw` function.
```

Comment (loicbourgois) [inline README.md]: > you just need to add the needed command after the definition of the new `scw` function

How do we do that exactly ? Can you provide an example ?

File: README.md [hunk 4]
```diff
@@ -112,10 +112,45 @@ Then, run the installation and remove the `.deb` file:
 ```bash
 dpkg -i /tmp/scw.deb && rm -f /tmp/scw.deb
 ```
+-->
 
 ## With a Docker Image
 
-For each release, we deliver a tagged image on the [Scaleway Docker Hub](https://hub.docker.com/r/scaleway/cli/tags) so can run `scw` in a sandboxed way: _Coming soon..._
+### Manual installation
```
Comment (loicbourgois) [inline README.md]: Is there a non manual way to install it with docker ? 

If so, maybe add the corresponding title before this section, with `Comming soo` as content.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+WORKDIR /go/src/github.com/scaleway/scaleway-cli
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+RUN go mod download
+
+COPY cmd/ cmd/
+COPY internal/ internal/
+COPY .git .git/
```
Comment (loicbourgois) [inline Dockerfile]: maybe

```suggestion
COPY .git/ .git/
```

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -a -ldflags "-w -extldflags -static -X main.GitCommit=$(git rev-parse --short HEAD) -X main.GitBranch=$(git symbolic-ref -q --short HEAD || echo HEAD) -X main.BuildDate=$(date -u '+%Y-%m-%dT%I:%M:%S%p')" ./cmd/scw
```
Comment (loicbourgois) [inline Dockerfile]: can we reuse the build script somehow ? 

https://github.com/scaleway/scaleway-cli/blob/v2/scripts/build.sh

would that necessitate lot of refactor ?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,21 @@
+FROM golang:1.14-alpine as builder
+
+RUN apk update && apk add --no-cache git ca-certificates && update-ca-certificates
```
Comment (loicbourgois) [inline Dockerfile]: can you explain in a comment why we need operations related to ca-certificates please ? 

it doesn't appear on the CLI v1, and it's not mentionned anywhere in the README of the v2

File: README.md [hunk 8]
```diff
@@ -112,16 +112,62 @@ Then, run the installation and remove the `.deb` file:
 ```bash
 dpkg -i /tmp/scw.deb && rm -f /tmp/scw.deb
 ```
+-->
 
 ## With a Docker Image
 
-For each release, we deliver a tagged image on the [Scaleway Docker Hub](https://hub.docker.com/r/scaleway/cli/tags) so can run `scw` in a sandboxed way: _Coming soon..._
+### Official releases (Comigng soon..)
```
Comment (remyleone) [inline README.md]:
```suggestion
### Official releases (Coming soon..)
```

File: Dockerfile (hunk 6 above)
Comment (Sh4d1) [reply Dockerfile]: Hmm everything is possible :smile: But we'll need to refactor the script that's for sure

Comment (Sh4d1) [reply Dockerfile]: done

File: scripts/build.sh [hunk 9]
```diff
@@ -15,9 +10,20 @@ LDFLAGS=(
...
 )
 
-VERSION=$(go run cmd/scw/main.go -o json version | jq -r .version | tr . -)
-
 export CGO_ENABLED=0
-GOOS=linux  GOARCH=amd64 go build -ldflags "${LDFLAGS[*]}" -o "$BIN_DIR/scw-$VERSION-linux-x86_64"  cmd/scw/main.go
-GOOS=darwin GOARCH=amd64 go build -ldflags "${LDFLAGS[*]}" -o "$BIN_DIR/scw-$VERSION-darwin-x86_64" cmd/scw/main.go
-GOOS=windows GOARCH=amd64 go build -ldflags "${LDFLAGS[*]}" -o "$BIN_DIR/scw-$VERSION-windows-x86_64.exe" cmd/scw/main.go
+
+if [[ "${1}" == "build-in-docker" ]]; then
```
Comment (loicbourgois) [inline scripts/build.sh]: what happens if 0 arg is passed ? does this fail ? 

isn't it better to have a flag that we set to `true` ?

Comment (Sh4d1) [reply scripts/build.sh]: if 0 args are passed `$1` is an empty string

File: .github/workflows/unit-tests.yaml [hunk 10]
```diff
@@ -41,3 +41,10 @@ jobs:
           fetch-depth: 1
       - name: Build binaries
         run: ./scripts/build.sh
+  docker-tests:
```
Comment (loicbourgois) [inline .github/workflows/unit-tests.yaml]:
```suggestion

  docker-tests:
```

File: README.md [hunk 11]
```diff
@@ -112,16 +112,62 @@ Then, run the installation and remove the `.deb` file:
...
+### Manual installation
+
+You can build the `scw` CLI with Docker. If you have Docker installed, you can run:
+```sh
+docker build -t scaleway/cli .
+```
+
+You can then use the CLI as you would run any image:
+```sh
+docker run -i scaleway/cli
```
Comment (jerome-quere) [inline README.md]: add `--rm` on all docker run or user will have many dead container

Comment (Sh4d1) [reply README.md]: fixed

File: README.md [hunk 12]
```diff
@@ -112,16 +112,62 @@ Then, run the installation and remove the `.deb` file:
...
+scw() {
+    docker run -i --rm -v $HOME/.config/scw/config.yaml:/.config/scw/config.yaml:ro scaleway/cli $@
+}
+```
+
+You can still use autocompletion while running the CLI in Docker, you just need to add the needed command after the definition of the new `scw` function.
+For instance, your `~/.bashrc` will look like this:
+```bash
+scw() {
+    docker run -i --rm -v $HOME/.config/scw/config.yaml:/.config/scw/config.yaml:ro scaleway/cli "$@"
```
Comment (jerome-quere) [inline README.md]: TODO: check if withiut `-t` we do not have probleme with terminal size detection

Comment (Sh4d1) [reply README.md]: done

**Comment by Sh4d1** [comment]
@QuentinBrosse hmm it's still pretty small :thinking:

File: docs/docker.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
+# Using the CLI with Docker
+
+The CLI can be used in Docker withou any problem.
```
Comment (QuentinBrosse) [inline docs/docker.md]:
```suggestion
The CLI can be used in Docker without any problem.
```

**Pull request merged**